FirebaseDynamicLinks/Tests/Unit/FIRDynamicLinksTest.m

```diff
@@ -894,6 +894,63 @@ - (void)testDynamicLinkFromUniversalLinkURLReturnsDLMinimumVersion {
   XCTAssertEqualObjects(expectedMinVersion, minVersion, @"Min version didn't match imv= parameter");
 }
 
+- (void)testDynamicLinkFromUniversalLinkURLReturnsUTMParams {
+  NSString *expectedUtmSource = @"utm_source";
+  NSString *expectedUtmMedium = @"utm_medium";
+  NSString *expectedUtmCampaign = @"utm_campaign";
+  NSString *expectedUtmTerm = @"utm_term";
+  NSString *expectedUtmContent = @"utm_content";
+
+  NSString *utmParamsString = [NSString
+      stringWithFormat:@"utm_source=%@&utm_medium=%@&utm_campaign=%@&utm_term=%@&utm_content=%@",
+                       expectedUtmSource, expectedUtmMedium, expectedUtmCampaign, expectedUtmTerm,
+                       expectedUtmContent];
+  NSString *urlSuffix =
+      [NSString stringWithFormat:@"%@&%@", kEncodedComplicatedURLString, utmParamsString];
+
+  NSString *urlString =
+      [NSString stringWithFormat:kStructuredUniversalLinkFmtSubdomainDeepLink, urlSuffix];
+  NSURL *url = [NSURL URLWithString:urlString];
+
+  [self.service setUpWithLaunchOptions:nil
+                                apiKey:kAPIKey
+                             urlScheme:kURLScheme
+                          userDefaults:self.userDefaults];
+  XCTestExpectation *expectation = [self expectationWithDescription:@"completion called"];
+  [self.service
+      dynamicLinkFromUniversalLinkURL:url
```

Comment:
Does one of these tests cover the escape that was caught by the 1P app?

Reply:
I mean - did you add a test so we don't need to rely on 1P tests to regression test?

Reply:
I see. Probably not necessary.Thanks.